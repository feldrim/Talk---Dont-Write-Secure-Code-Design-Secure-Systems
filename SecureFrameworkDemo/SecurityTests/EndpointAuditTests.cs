using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace SecurityTests {

	[TestClass]
	public class EndpointAuditTests {

		[TestMethod]
		[UseReporter(typeof(DiffReporter))]
		public void Endpoint_analysis_file_matches_expectations() {
			var auditReport = SecurityAudit.Program.GenerateReport();
			
			Approvals.Verify(auditReport);
		}
	}
}
