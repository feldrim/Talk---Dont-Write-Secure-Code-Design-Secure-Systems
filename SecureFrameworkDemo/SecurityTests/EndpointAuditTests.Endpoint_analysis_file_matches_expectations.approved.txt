AutoLogin/AutoLogin
PUBLIC / [no perm required]

Home/About
PUBLIC / [no perm required]

Home/Contact
PUBLIC / [no perm required]

Home/Index
PUBLIC / [no perm required]

Insecure/Index
PUBLIC / [no perm required]

Insecure/ManageOrders
PUBLIC / [no perm required]

Insecure/ModifyOrder
PUBLIC / [no perm required]

Insecure/ModifyOrderAjax
PUBLIC / [no perm required]

Insecure/OrderDetail
PUBLIC / [no perm required]

Insecure/OrderList
PUBLIC / [no perm required]

SecureFeature/Index
PUBLIC / [no perm required]

SecureFeature/ManageOrders
PUBLIC / [no perm required]

SecureFeature/ModifyOrder
PUBLIC / [no perm required]

SecureFeature/ModifyOrderAjax
PUBLIC / [no perm required]

SecureFeature/OrderDetail
PUBLIC / [no perm required]

SecureFeature/OrderList
PUBLIC / [no perm required]

SecureFramework/Index
REQUIRES AUTH / [no perm required]

SecureFramework/ManageOrders
REQUIRES AUTH / ManageOrders

SecureFramework/ModifyOrder
REQUIRES AUTH / [no perm required]

SecureFramework/ModifyOrderAjax
REQUIRES AUTH / [no perm required]

SecureFramework/OrderDetail
REQUIRES AUTH / [no perm required]

SecureFramework/OrderList
REQUIRES AUTH / [no perm required]

SecureFramework/SomeMethodThatShouldBePrivate
REQUIRES AUTH / [no perm required]

SecurityAudit/Index
PUBLIC / [no perm required]

